src/producer.rs

```diff
@@ -0,0 +1,78 @@
+use crate::{SharedFluvio, DEFAULT_TOPIC};
+
+use log::debug;
+use fluvio::TopicProducer;
+use fluvio::FluvioError;
+
+use node_bindgen::derive::node_bindgen;
+use node_bindgen::core::NjError;
+use node_bindgen::core::val::JsEnv;
+use node_bindgen::core::TryIntoJs;
+use node_bindgen::sys::napi_value;
+use node_bindgen::core::JSClass;
+
+pub struct TopicProducerWrapper {
+    client: SharedFluvio,
```

Comment:
Why is this wrapping SharedFluvio?  It should be wrapping TopicProducer.   I see reason that's why it's using SharedFluvio.    This is incorrect.

Reply:
commit a1cd4c now uses the producer and consumer clients directly. I have not tested these changes against the test suite; will do that in the subsequent PR.